spec_integration/facter_block_legacy_spec.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require 'json'
+require_relative 'integration_helper'
+
+describe 'Facter' do
+  context 'when legacy facts is blocked' do
+    let(:facter_conf_content) do
+      'facts : {
+  blocklist : [ "legacy" ],
+}'
+    end
+
+    facter_output = ''
+
+    before do
+      facter_conf_path = IntegrationHelper.create_file('./temp/facter.conf', facter_conf_content)
+
+      out, = IntegrationHelper.exec_facter('-c', facter_conf_path, '--show-legacy', '-j')
+      facter_output = JSON.parse(out)
+    end
```

Comment:
this can be replaced using `let`, which is lazy initialised
```ruby
let(:facter_output) do
  facter_conf_path = IntegrationHelper.create_file('./temp/facter.conf', facter_conf_content)
  out, = IntegrationHelper.exec_facter('-c', facter_conf_path, '--show-legacy', '-j')
  JSON.parse(out)
end
```